pkg/apiserver/filters/impersonation.go

```diff
@@ -123,6 +123,15 @@ func WithImpersonation(handler http.Handler, requestContextMapper api.RequestCon
 		oldUser, _ := api.UserFrom(ctx)
 		httplog.LogOf(req, w).Addf("%v is acting as %v", oldUser, newUser)
 
+		// clear all the impersonation headers from the request
+		req.Header.Del(authenticationapi.ImpersonateUserHeader)
```

Comment:
> Isnt front proxy just a user of the generic "impersonation framework"?

The front proxy uses a different mechanism.  The proxy is assured of being able to specify any user, so no additional authentication check is performed.  In addition, you can use the front proxy to identify the user on the request and that user can ask to impersonate someone else.  The features are similar and certainly complementary, but not identical.

Having gone through the process of creating an API proxy that terminates authentication, I think we want to use the auth-proxy mechanism instead of impersonation like I first thought.